### PR TITLE
add language detection support for azure

### PR DESCRIPTION
**Pull request by GS-BLaszlo**
contribution for: 
[this issue](https://github.com/OvidijusParsiunas/speech-to-element/issues/5)

### CODE REVIEW

File: component/index.html [hunk 1]
```diff
@@ -1,57 +1,58 @@
...
-        //   },
-        //   onError: () => {
-        //     errorEl.style.display = 'block';
-        //     buttonEl.innerText = 'Start';
-        //   },
-        // });
-      };
-    </script>
-  </body>
+<head>
```
Comment (OvidijusParsiunas) [inline component/index.html]: Can you undo the changes below as they contain sample configuration to help developers get started. Thankyou.

File: component/src/services/azure/azure.ts [hunk 2]
```diff
@@ -52,7 +52,13 @@ export class Azure extends Speech {
     const audioConfig = speechSDK.AudioConfig.fromDefaultMicrophoneInput();
     const speechConfig = await AzureSpeechConfig.get(speechSDK.SpeechConfig, options);
     if (speechConfig) {
-      const recognizer = new speechSDK.SpeechRecognizer(speechConfig, audioConfig);
+      const recognizer = speechSDK.SpeechRecognizer.FromConfig(
```
Comment (OvidijusParsiunas) [inline component/src/services/azure/azure.ts]: The original config (`const recognizer = new speechSDK.SpeechRecognizer(speechConfig, audioConfig);`) should still be used when `autoLanguages` is not defined.
If `autoLanguages` is defined use your implementation then.

File: component/src/types/options.ts [hunk 3]
```diff
@@ -56,6 +56,8 @@ export interface AzureOptions {
   region: string;
   // https://docs.microsoft.com/azure/cognitive-services/speech-service/supported-languages
   language?: string;
+  // max 4 languages can be added (azure limitation)
```
Comment (OvidijusParsiunas) [inline component/src/types/options.ts]: Can you point me towards the documentation that states that max is 4? Thankyou!

File: component/src/types/options.ts [hunk 4]
```diff
@@ -56,6 +56,8 @@ export interface AzureOptions {
...
+  autoLanguages?: string[];
```
Comment (OvidijusParsiunas) [inline component/src/types/options.ts]: Please use the following type which has the constraint mentioned above:
`[string, ...string[]] & {length: 1 | 2 | 3 | 4}`
Thankyou!

**Review by OvidijusParsiunas — CHANGES_REQUESTED** [review]

File: component/src/types/options.ts (hunk 3 above)
Comment (GS-BLaszlo) [reply component/src/types/options.ts]: if we try to use 5 languages, an error will appear:
`Language identification only supports 4 languages in DetectAudioAtStart mode websocket error code: 1007`
Documentation:
https://learn.microsoft.com/en-us/azure/ai-services/speech-service/language-identification?tabs=once&pivots=programming-language-javascript

> You can include up to four languages for [at-start LID](https://learn.microsoft.com/en-us/azure/ai-services/speech-service/language-identification?tabs=once&pivots=programming-language-javascript#at-start-and-continuous-language-identification) or up to 10 languages for [continuous LID](https://learn.microsoft.com/en-us/azure/ai-services/speech-service/language-identification?tabs=once&pivots=programming-language-javascript#at-start-and-continuous-language-identification).

Comment (GS-BLaszlo) [reply component/src/types/options.ts]: I updated the code now we can use up to 10 if in `Continuous` detection mode

**Comment by GS-BLaszlo** [comment]
@OvidijusParsiunas pls approve

**Comment by GS-BLaszlo** [comment]
@OvidijusParsiunas 
I have updated all the code, i'm dire need in this modification

**Review by OvidijusParsiunas — APPROVED** [review]

**Comment by OvidijusParsiunas** [comment]
Hi apologies, for the long wait. I maintain multiple projects and recently have had very little free time, hence everything has been delayed. Your code looks good, I have merged it and published it to `speech-to-element` version `1.0.2`.
